1. Faça um Programa que peça dois números e imprima o maior deles.

In [2]:
n1 = int(input("Digite o primeiro numero:"))
n2 = int(input("Digite o segundo numero:"))
if n1 > n2:
    print ("O maior numero é",n1)
else:
    print ("O maior numero é",n2)

Digite o primeiro numero:2
Digite o segundo numero:6
O maior numero é 6


2. Faça um Programa que verifique se uma letra digitada é vogal ou consoante.

In [6]:
letra = str(input("Digite uma letra:"))
if letra == "a" or letra == "e" or letra == "i" or letra == "o" or letra == "u" or letra == "A" or letra == "E" or letra == "I" or letra == "O" or letra == "U":
    print ("Vogal")
else:
    print("Consoante")

Digite uma letra:i
Vogal


3. Faça um programa para a leitura de duas notas parciais de um aluno. O programa
deve calcular a média alcançada por aluno e apresentar:
-A mensagem &quot;Aprovado&quot;, se a média alcançada for maior ou igual a sete;
-A mensagem &quot;Reprovado&quot;, se a média for menor do que sete;
-A mensagem &quot;Aprovado com Distinção&quot;, se a média for igual a dez.

In [9]:
nota1 = float(input("Digite a primeira nota:"))
nota2 = float(input("Digite a segunda nota:"))
media = (nota1 + nota2)/2
print ("A media foi",media)
if media >= 7:
    print ("Aprovado")
elif media > 10:
    print ("Aprovado com Distinção")
else:
    print ("Reprovado")

Digite a primeira nota:10
Digite a segunda nota:10
A media foi 10.0
Aprovado


4. Faça um Programa que leia três números e mostre-os em ordem decrescente.

In [16]:
n1 = int(input("Digite o primeiro numero:"))
n2 = int(input("Digite o segundo numero:"))
n3 = int(input("Digite o terceiro numero:"))
if   n1 > n2 and n2 > n3:
    print("A ordem dos numeros é",n3 , n2 , n1)
elif n2 > n3 and n3 > n1:
    print("A ordem dos numeros é",n1 , n3 , n2)
elif n3 > n1 and n1 > n2:
    print("A ordem dos numeros é",n2 , n1 , n3)
else:
    print("A ordem dos numeros é",n3 , n2 , n1)

Digite o primeiro numero:2
Digite o segundo numero:6
Digite o terceiro numero:9
A ordem dos numeros é 9 6 2


5. As Organizações Tabajara resolveram dar um aumento de salário aos seus
colaboradores e lhe contrataram para desenvolver o programa que calculará os
reajustes.

Faça um programa que recebe o salário de um colaborador e o reajuste segundo o
seguinte critério, baseado no salário atual:

-salários até R$ 280,00 (incluindo) : aumento de 20%

-salários entre R$ 280,00 e R$ 700,00 : aumento de 15%

-salários entre R$ 700,00 e R$ 1500,00 : aumento de 10%

-salários de R$ 1500,00 em diante : aumento de 5% Após o aumento ser realizado

informe na tela:
-o salário antes do reajuste;
-o percentual de aumento aplicado;
-o valor do aumento;
-o novo salário, após o aumento.

In [27]:
salario = int(input("Digite o seu salario:"))
if salario <= 280:
    novosal = ((salario/100)*20) + salario
    reajP = "20%"
    reajS = (salario/10)*20
elif salario > 280 and salario <= 700:
    novosal = ((salario/100)*15) + salario
    reajP = "15%"
    reajS = (salario/100)*15
elif salario > 700 and salario <= 1500:
    novosal = ((salario/100)*10) + salario
    reajP = "10%"
    reajS = (salario/100)*10
else:
    novosal = ((salario/100)*5) +salario
    reajP = "5%"
    reajS = (salario/100)*5
print ("O salario antes do reajuste é de R$",salario)
print ("O percentual de reajuste é de", reajP)
print ("O valor do aumento é de R$",reajS,)
print ("Seu novo salario será de R$",novosal)
    

Digite o seu salario:1200
O salario antes do reajuste é de R$ 1200
O percentual de reajuste é de 10%
O valor do aumento é de R$ 120.0
Seu novo salario será de R$ 1320.0


6. Faça um Programa que leia um número e exiba o dia correspondente da semana.
(1-Domingo, 2- Segunda, etc.), se digitar outro valor deve aparecer valor inválido.

In [29]:
dia = int(input("Digite um numero para escolher o dia da semana: "))
if dia == 1:
    print("Domingo")
elif dia == 2:
    print("segunda")
elif dia == 3:
    print("terça")
elif dia == 4:
    print("Quarta")
elif dia == 5:
    print("Quinta")
elif dia == 6:
    print("Sexta")
elif dia == 7:
    print("Sabado")
else:
    print("Dia não encontrado")


Digite um numero para escolher o dia da semana: 6
Sexta


7. Faça um programa que lê as duas notas parciais obtidas por um aluno numa
disciplina ao longo de um semestre, e calcule a sua média. A atribuição de conceitos
obedece à tabela abaixo:

Média de Aproveitamento Conceito

Entre 9.0 e 10.0 A

Entre 7.5 e 9.0 B

Entre 6.0 e 7.5 C

Entre 4.0 e 6.0 D

Entre 4.0 e zero E

O algoritmo deve mostrar na tela as notas, a média, o conceito correspondente e a
mensagem “APROVADO” se o conceito for A, B ou C ou “REPROVADO” se o conceito
for D ou E.

In [33]:
nota1 = float(input("Digite a primeira nota: "))
nota2 = float(input("Digite a segunda nota: "))
media = (nota1 + nota2)/2
print("----------------------------------------------")
if media >= 9.0:
    print("A primeira nota foi", nota1)
    print("A segunda nota foi", nota2)
    print("A media entre as duas notas foi de", media)
    print("Você tirou um A!")
    print("APROVADO")
elif media >= 7.5 and media < 9:
    print("A primeira nota foi", nota1)
    print("A segunda nota foi", nota2)
    print("A media entre as duas notas foi de", media)
    print("Você tirou um B!")
    print("APROVADO")
elif media >= 6.0 and media < 7:
    print("A primeira nota foi", nota1)
    print("A segunda nota foi", nota2)
    print("A media entre as duas notas foi de", media)
    print("Você tirou um C!")
    print("APROVADO")
elif media >= 4.0 and media < 6:
    print("A primeira nota foi", nota1)
    print("A segunda nota foi", nota2)
    print("A media entre as duas notas foi de", media)
    print("Você tirou um D!")
    print("REPROVADO")
elif media >= 0 and media < 4:
    print("A primeira nota foi", nota1)
    print("A segunda nota foi", nota2)
    print("A media entre as duas notas foi de", media)
    print("Você tirou um E!")
    print("REPROVADO")

    

Digite a primeira nota: 9.0
Digite a segunda nota: 8.5
----------------------------------------------
A primeira nota foi 9.0
A segunda nota foi 8.5
A media entre as duas notas foi de 8.75
Você tirou um B!
APROVADO


8. Faça um Programa que peça os 3 lados de um triângulo. O programa deverá
informar se os valores podem ser um triângulo. Indique, caso os lados formem um
triângulo, se o mesmo é: equilátero, isósceles ou escaleno.
Dicas:
-Três lados formam um triângulo quando a soma de quaisquer dois lados for maior
que o terceiro;
-Triângulo Equilátero: três lados iguais;
-Triângulo Isósceles: quaisquer dois lados iguais;
-Triângulo Escaleno: três lados diferentes;

In [35]:
lado1 = float(input("Digite o primeiro lado do triangulo: "))
lado2 = float(input("Digite o segundo lado do triangulo: "))
lado3 = float(input("Digite o terceiro lado do triangulo: "))
if lado1 > (lado2 + lado3) or lado2 > (lado3 + lado1) or lado3 > (lado1 + lado2):
    print("Os valores indicados não podem formar um triangulo")
elif lado1 == lado2 == lado3:
    print("Equilátero")
elif lado1 == lado2 or lado1 == lado3 or lado2 == lado3:
    print("Isósceles")
else:
    print("Escaleno")

Digite o primeiro lado do triangulo: 7.0
Digite o segundo lado do triangulo: 7.0
Digite o terceiro lado do triangulo: 3.0
Isósceles


9. Faça um programa que calcule as raízes de uma equação do segundo grau, na
forma ax2 + bx + c. O programa deverá pedir os valores de a, b e c e fazer as
consistências, informando ao usuário nas seguintes situações:

 Se o usuário informar o valor de A igual a zero, a equação não é do segundo grau e
o programa não deve fazer pedir os demais valores, sendo encerrado;

 Se o delta calculado for negativo, a equação não possui raizes reais. Informe ao
usuário e encerre o programa;

 Se o delta calculado for igual a zero a equação possui apenas uma raiz real;
informe-a ao usuário;

 Se o delta for positivo, a equação possui duas raiz reais; informe-as ao usuário;

In [38]:
import math
a = int(input("Digite o A da equeção: "))
b = int(input("Digite o B da equeção: "))
c = int(input("Digite o C da equeção: "))
if a == 0:
    print("a equação não é do segundo grau")
else:
    delta = b*b - (4*a*c)
if delta < 0:
    print(" a equação não possui raizes reais")
elif delta == 0:
    raiz = -b / (2*a)
    print("Delta = 0 e a raiz é", raiz)
else:
    raiz1 = (-b + math.sqrt(delta) ) / (2*a)
    raiz2 = (-b - math.sqrt(delta) ) / (2*a)    
    print('Raizes: ',raiz1,' e ',raiz2)

Digite o A da equeção: 2
Digite o B da equeção: 6
Digite o C da equeção: 3
Raizes:  -0.6339745962155614  e  -2.3660254037844384


10. Faça um Programa para um caixa eletrônico. O programa deverá perguntar ao
usuário a valor do saque e depois informar quantas notas de cada valor serão
fornecidas. As notas disponíveis serão as de 1, 5, 10, 50 e 100 reais. O valor mínimo é
de 10 reais e o máximo de 600 reais. O programa não deve se preocupar com a
quantidade de notas existentes na máquina.

Exemplo 1: Para sacar a quantia de 256 reais, o programa fornece duas notas de
100, uma nota de 50, uma nota de 5 e uma nota de 1;

Exemplo 2: Para sacar a quantia de 399 reais, o programa fornece três notas de
100, uma nota de 50, quatro notas de 10, uma nota de 5 e quatro notas de 1.

In [41]:
numero = int(input("Digite quanto você quer sacar: "))
cem = int(numero / 100)
numero = numero % 100    
cinq = int(numero/50)
numero = numero % 50
dez = int(numero/10)
numero = numero % 10
cinco = int(numero/5)
numero = numero % 5
um = numero
    
print('Notas R$100,00 = ',cem)
print('Notas R$ 50,00 = ',cinq)
print('Notas R$ 10,00 = ',dez)
print('Notas R$  5,00 = ',cinco)
print('Notas R$  1,00 = ',um)

Digite quanto você quer sacar: 560
Notas R$100,00 =  5
Notas R$ 50,00 =  1
Notas R$ 10,00 =  1
Notas R$  5,00 =  0
Notas R$  1,00 =  0


11. Faça um programa que faça 5 perguntas para uma pessoa sobre um crime. As
perguntas são:
&quot;Telefonou para a vítima?&quot;
&quot;Esteve no local do crime?&quot;
&quot;Mora perto da vítima?&quot;
&quot;Devia para a vítima?&quot;
&quot;Já trabalhou com a vítima?&quot;
O programa deve no final emitir uma classificação sobre a participação da pessoa no
crime. Se a pessoa responder positivamente a 2 questões ela deve ser classificada
como &quot;Suspeita&quot;, entre 3 e 4 como &quot;Cúmplice&quot; e 5 como &quot;Assassino&quot;. Caso contrário,
ele será classificado como &quot;Inocente&quot;.

In [15]:
count = 0
perg = str(input("Você telefonou para a vitima ?"))
perg.lower()
if perg == "sim" or perg == "s":
    count = count + 1
perg = str(input("Você esteve no local do crime ?"))
perg.lower()
if perg == "sim" or perg == "s":
    count = count + 1
perg = str(input("Você mora perto da vitima ?"))
perg.lower()
if perg == "sim" or perg == "s":
    count = count + 1
perg = str(input("Você devia para a vitima ?"))
perg.lower()
if perg == "sim" or perg == "s":
    count = count + 1
perg = str(input("Você já trabalhou com a vitima ?"))
perg.lower()
if perg == "sim" or perg == "s":
    count = count + 1
if count == 2:
    print ("Suspeito(a)")
if count == 3 or count == 4:
    print ("Cumplice") 
if count == 5:
    print ("Assassino")
else:
    print ("Inocente")

Você telefonou para a vitima ?s
Você esteve no local do crime ?n
Você mora perto da vitima ?n
Você devia para a vitima ?n
Você já trabalhou com a vitima ?n
Inocente


12 Uma fruteira está vendendo frutas com a seguinte tabela de preços:

            Até 5 Kg        Acima de 5 Kg   
Morango     R$ 2,50 por     Kg R$ 2,20 por Kg   
Maçã        R$ 1,80 por     Kg R$ 1,50 por Kg    

Se o cliente comprar mais de 8 Kg em frutas ou o valor total da compra ultrapassar R$ 25,00, receberá ainda um desconto de 10% sobre este total. Escreva um algoritmo para ler a quantidade (em Kg) de morangos e a quantidade (em Kg) de maças adquiridas e escreva o valor a ser pago pelo cliente.

In [1]:
def preco_morango(kilos):
    if(kilos < 5):
        return kilos * 2.50
    else:
        return kilos * 2.20

#funcao que calcula o valor do preco da maca
def preco_maca(kilos):
    if(kilos < 5):
        return kilos * 1.80
    else:
        return kilos * 1.50

#funcao que retorna uma soma
def soma(morango, maca):
    return morango + maca

morango = float(input("Digite quantos kilos de morango:"))
maca = float(input("Digite quantos kilos de maçã: "))


preco_mor = preco_morango(morango)
preco_mac = preco_maca(maca)

if(soma(morango, maca) >= 8 or soma(preco_mor, preco_mac) >= 25):
    desconto = 0.10
    preco_fim = soma(preco_mor, preco_mac) * (1 - desconto)
else:
    desconto = 0.00
    preco_fim = soma(preco_mor, preco_mac)
    print(
     f'Quantidade de Morangos - KG: {morango} \tPreco: R$ {preco_mor} \n'
     + f'Quantidade de Maçãs    - KG: {maca} \tPreco: R$ {preco_mac} \n'
     + f'Preço total dos dois:  R$ {preco_mac + preco_mor} \n'
     + 'Desconto: %d %% \n' %(desconto*100)
     + 'Preço final da compra: R$ %.2f'%(preco_fim))

Digite quantos kilos de morango:5
Digite quantos kilos de maçã: 7


13 Faça um programa que peça uma nota, entre zero e dez. Mostre uma mensagem caso o valor seja inválido e continue pedindo até que o usuário informe um valor válido.

In [2]:
nota = float(input("Digite uma nota de 0 a 10: "))

while nota > 10 or nota < 0:
    print('Valor inválido')
    nota = float(input('Digite uma nota de "0 a 10": '))

print(f'Nota: {nota}')

Digite uma nota de 0 a 10: 8
Nota: 8.0


14 Faça um programa que leia 5 números e informe o maior número.

In [3]:
numeros = [int(input(f'Digite o {x+1}º numero: ')) for x in range(5)]

print('Os numeros digitados foram:',numeros)
print(f'O maior número foi: {max(numeros)}')

Digite o 1º numero: 3
Digite o 2º numero: 5
Digite o 3º numero: 8
Digite o 4º numero: 9
Digite o 5º numero: 0
Os numeros digitados foram: [3, 5, 8, 9, 0]
O maior número foi: 9


15 Faça um programa que leia 5 números e informe a soma e a média dos números.

In [4]:
numeros = [int(input(f'Digite o {x+1}º numero: ')) for x in range(5)]

soma = sum(numeros)

print('Os numeros digitados foram:',numeros)
print('A SOMA dos numeros:', soma)
print('A MEDIA dos numeros:', soma / 5)

Digite o 1º numero: 2
Digite o 2º numero: 5
Digite o 3º numero: 7
Digite o 4º numero: 8
Digite o 5º numero: 6
Os numeros digitados foram: [2, 5, 7, 8, 6]
A SOMA dos numeros: 28
A MEDIA dos numeros: 5.6


16 Faça um programa que imprima na tela apenas os números ímpares entre 1 e 50.

In [5]:
print([x for x in range(1, 51)][::2])


[1, 3, 5, 7, 9, 11, 13, 15, 17, 19, 21, 23, 25, 27, 29, 31, 33, 35, 37, 39, 41, 43, 45, 47, 49]


17 Desenvolva um gerador de tabuada, capaz de gerar a tabuada de qualquer número inteiro entre 1 a 10. O usuário deve informar de qual numero ele deseja ver a tabuada. A saída deve ser conforme o exemplo abaixo:

Tabuada de 5:
5 X 1 = 5
5 X 2 = 10
...
5 X 10 = 50

In [6]:
tabuada = int(input("Insira o número da tabuada: "))
for mt in range(1, 11):
    res = mt*tabuada
    print(f"{tabuada} X {mt} = {res}")

Insira o número da tabuada: 2
2 X 1 = 2
2 X 2 = 4
2 X 3 = 6
2 X 4 = 8
2 X 5 = 10
2 X 6 = 12
2 X 7 = 14
2 X 8 = 16
2 X 9 = 18
2 X 10 = 20


18 A série de Fibonacci é formada pela seqüência 1,1,2,3,5,8,13,21,34,55,... Faça um programa capaz de gerar a série até o n−ésimo termo.

In [8]:
a = 0
b = 1
nesimo = int(input("Digite até qual termo a série será gerada:"))
for i in range(nesimo):
    print(a)
    aux = b
    b = a + b
    a = aux

Digite até qual termo a série será gerada:6
0
1
1
2
3
5


19 Faça um programa que calcule o fatorial de um número inteiro fornecido pelo usuário.
Ex.: 5!=5.4.3.2.1=120

In [9]:
fat_num = int(input("Insira um número: "))
fat_numero = fat_num
numeros = []
fatorial = 1

while(fat_num > 0):
    numeros.append(fat_num)
    fat_num -= 1

for n in numeros:
    fatorial *= n

print(f'O fatorial de {fat_numero} é:')
print (f'{fat_numero}! = {numeros} = {fatorial}')

Insira um número: 3
O fatorial de 3 é:
3! = [3, 2, 1] = 6


20 O Sr. Manoel Joaquim possui uma grande loja de artigos de R$ 1,99, com cerca de 10 caixas. Para agilizar o cálculo de quanto cada cliente deve pagar ele desenvolveu um tabela que contém o número de itens que o cliente comprou e ao lado o valor da conta. Desta forma a atendente do caixa precisa apenas contar quantos itens o cliente está levando e olhar na tabela de preços. Você foi contratado para desenvolver o programa que monta esta tabela de preços, que conterá os preços de 1 até 50 produtos, conforme o exemplo abaixo:

Lojas Quase Dois - Tabela de preços
1 - RS 1.99
2 - RS 3.98
...
50 - RS 99.50

In [10]:
print('Lojas Quase Dois - Tabela de preços')
for i in range(1, 51):
    preco = i * 1.99
    print(f'{i} - R$ {preco:.2f}')

Lojas Quase Dois - Tabela de preços
1 - R$ 1.99
2 - R$ 3.98
3 - R$ 5.97
4 - R$ 7.96
5 - R$ 9.95
6 - R$ 11.94
7 - R$ 13.93
8 - R$ 15.92
9 - R$ 17.91
10 - R$ 19.90
11 - R$ 21.89
12 - R$ 23.88
13 - R$ 25.87
14 - R$ 27.86
15 - R$ 29.85
16 - R$ 31.84
17 - R$ 33.83
18 - R$ 35.82
19 - R$ 37.81
20 - R$ 39.80
21 - R$ 41.79
22 - R$ 43.78
23 - R$ 45.77
24 - R$ 47.76
25 - R$ 49.75
26 - R$ 51.74
27 - R$ 53.73
28 - R$ 55.72
29 - R$ 57.71
30 - R$ 59.70
31 - R$ 61.69
32 - R$ 63.68
33 - R$ 65.67
34 - R$ 67.66
35 - R$ 69.65
36 - R$ 71.64
37 - R$ 73.63
38 - R$ 75.62
39 - R$ 77.61
40 - R$ 79.60
41 - R$ 81.59
42 - R$ 83.58
43 - R$ 85.57
44 - R$ 87.56
45 - R$ 89.55
46 - R$ 91.54
47 - R$ 93.53
48 - R$ 95.52
49 - R$ 97.51
50 - R$ 99.50
